src/test/java/de/zalando/aruha/nakadi/controller/EventStreamControllerTest.java

```diff
@@ -0,0 +1,89 @@
+package de.zalando.aruha.nakadi.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import de.zalando.aruha.nakadi.config.NakadiConfig;
+import de.zalando.aruha.nakadi.repository.TopicRepository;
+import de.zalando.aruha.nakadi.service.EventStream;
+import de.zalando.aruha.nakadi.service.EventStreamConfig;
+import de.zalando.aruha.nakadi.service.EventStreamFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.springframework.http.converter.StringHttpMessageConverter;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.standaloneSetup;
+
+public class EventStreamControllerTest {
+
+    private static final String TEST_EVENT_TYPE = "test";
+
+    private final ObjectMapper objectMapper = new NakadiConfig().jacksonObjectMapper();
+
+    private final MappingJackson2HttpMessageConverter messageConverter =
+            new MappingJackson2HttpMessageConverter(objectMapper);
+
+    private MockMvc mockMvc;
+
+    private TopicRepository topicRepositoryMock;
+
+    private EventStreamFactory eventStreamFactoryMock;
+
+    @Before
+    public void setup() {
+        topicRepositoryMock = mock(TopicRepository.class);
+        eventStreamFactoryMock = mock(EventStreamFactory.class);
+
+        final EventStreamController controller = new EventStreamController(topicRepositoryMock, objectMapper,
+                eventStreamFactoryMock);
+
+        mockMvc = standaloneSetup(controller)
+                .setMessageConverters(new StringHttpMessageConverter(), messageConverter)
+                .build();
+    }
+
+    @Test
+    public void whenNoParamsThenDefaultsAreUsed() throws Exception {
```

Comment:
This is basically all that I could test with mockMvc here.
The problem is that there's no a good way to write unit test for async endpoint. For example, mockMvc always finishes before the body is started to stream and before all checks are done, so it's not possible to check status code or response with it.